source/local/local.go

```diff
@@ -318,6 +321,47 @@ func getFileContent(fileName string) ([][]byte, error) {
 	return lines, nil
 }
 
+func (s *localSource) runNotifier(ch chan struct{}) {
+	for {
+		select {
+		case event := <-s.fsWatcher.Events:
+			if event.Op&fsnotify.Create == fsnotify.Create || event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Remove == fsnotify.Remove || event.Op&fsnotify.Rename == fsnotify.Rename || event.Op&fsnotify.Chmod == fsnotify.Chmod {
+				klog.InfoS("fsnotify event", event)
+				ch <- struct{}{}
+			}
+		case err := <-s.fsWatcher.Errors:
+			klog.ErrorS(err, "failed to to watch features.d changes")
+		}
+	}
+}
+
+// SetChannel method of the EventSource Interface
+func (s *localSource) SetChannel(ch chan struct{}) error {
+	info, err := os.Stat(featureFilesDir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	if info != nil && info.IsDir() {
```

Comment:
nit: maybe we should also check if `s.fsWatcher` is not nil? Kinda theoretical but thinking if this would be called multiple times